lib/SILGen/SILGenProlog.cpp

```diff
@@ -1382,20 +1382,28 @@ void SILGenFunction::emitProlog(
 
   emitExpectedExecutor();
 
-  // IMPORTANT: This block should be the last one in `emitProlog`,
-  // since it terminates BB and no instructions should be insterted after it.
-  // Emit an unreachable instruction if a parameter type is
-  // uninhabited
-  if (paramList) {
-    for (auto *param : *paramList) {
-      if (param->getTypeInContext()->isStructurallyUninhabited()) {
-        SILLocation unreachableLoc(param);
-        unreachableLoc.markAsPrologue();
-        B.createUnreachable(unreachableLoc);
-        break;
-      }
-    }
-  }
+  // Emits an unreachable instruction if a parameter type (including `self`) is
+  // uninhabited. Returns `true` if an unreachable location was created.
+  auto handleUninhabitedParam = [&](ParamDecl *param) -> bool {
+    if (!param->getTypeInContext()->isStructurallyUninhabited())
+      return false;
+
+    SILLocation unreachableLoc(param);
+    unreachableLoc.markAsPrologue();
+    B.createUnreachable(unreachableLoc);
+    return true;
+  };
+
+  // IMPORTANT: An unreachable block should be the last one in `emitProlog`,
+  // since it is a terminator and no instructions should be insterted after it.
+  if (selfParam)
```

Comment:
with this change, at least one part of the standard library now emits a warning when built – the `Never` [conformance](https://github.com/natecook1000/swift/blob/869fc1792e8967227e84e583a2f529ecb5ce29ea/stdlib/public/core/Policy.swift#L41) to `Identifiable`. should that be changed? i think if [this change](https://github.com/swiftlang/swift/pull/75574) is also merged, then the implementation could perhaps just be removed.